src/QMCHamiltonians/OperatorBase.h

```diff
@@ -246,6 +246,7 @@ struct OperatorBase : public QMCTraits
    *@return the value of the Hamiltonian component
    */
   virtual Return_t evaluate(ParticleSet& P) = 0;
+  virtual Return_t evaluateDeterministic(ParticleSet& P);
```

Comment:
Documentation is needed for this function. Please describe the desired behaviour.

Reply:
Could you say a bit more about waht "deterministically" means?